lib/buildFactory.js

```diff
@@ -265,6 +269,16 @@ class BuildFactory extends BaseFactory {
                     ];
 
                     modelConfig.templateId = job.templateId;
+                    if (job.templateId !== undefined && job.templateId !== null) {
+                        Promise.all([
+                            templateFactory.get({ id: job.templateId }),
+                            pipelineFactory.get({ id: job.pipelineId })
+                        ]).then(([t, p]) =>
+                            logger.info(
+                                `pipelineId:${p.id}: ${p.name} uses template ${t.namespace}/${t.name}@${t.version}`
```

Comment:
We are already storing build count in DB and is available for each template. For eg: https://cd.screwdriver.cd/templates/python/validate_type

You don't even need to do batch processing in DB, this can be done at API level. Why add extra DB calls and logs for something which is already built in ?

Reply:
There are three reasons.

1. Our platform has a general-purpose mechanism that measures the number of occurrences from logs and graphs them. Using this is the most efficient way to get the aggregate results I want.
2. There are many templates prepared in YJ. I would like to investigate which template is used frequently and whether the number of users is increasing. To get usage statistics for multiple templates, we need to create a slightly more sophisticated system. If OSS concludes that the output to the log is not valid, then we(YJ) intends to do this.
3. I can also think that logging the template uses used is reasonable

Reply:
Number of jobs and builds which uses a template is already stored with template, so we can already query this in an offline flow.  

We can keep the log, but please hide it behind a feature flag. Logging is not the problem, but the 2 extra DB calls done for each build is. So I don't want this to be default but controlled by a feature flag. Let's say `SD_DEBUG_LOGS` or something. 

This is not the same as [LOG_LEVEL](https://github.com/screwdriver-cd/logger/blob/master/index.js#L6), but we are using this logging to provide extra features in a build cluster. I would prefer to have SD cluster admins to have control over whether this extra DB calls is needed or not.

Reply:
I've been convinced right now that getting the data from the DB or API is the most reasonable way. So I want to close this PR without merging.

I understood the idea that `SD_DEBUG_LOGS` controls whether or not to log when more information is required even if the performance deteriorates. Certainly it looks good. However, when more logs are controlled by this flag in the future, I will have to turn this flag off for better performance. It will be a temporary solution to its original purpose, but it will not be permanent for me. When I need to turn the flag off, I'll change the aggregation mechanism to retrieve the data from the DB or API. It makes the most sense to get the data from the DB or API from the beginning.

Thanks, your opinion.